lib/rules/arrow-body-style.js

```diff
@@ -109,10 +109,22 @@ module.exports = {
                 }
 
                 if (never || asNeeded && blockBody[0].type === "ReturnStatement") {
+                    let message;
+
+                    if (blockBody.length === 0) {
+                        message = "Unexpected block statement surrounding arrow body; put `undefined` immediately after the `=>`.";
```

Comment:
I'm not sure I like this message because `undefined` isn't a keyword and some folks might prefer `void 0` or some other equivalent. Is there a way we can improve it? (I haven't got anything concrete now, sorry.)

Reply:
Would something like "move a value of `undefined` right after the `=>`" be better?

Alternatively, we could just use the old message for this case rather than adding an additional clause.

Personally, I think that:

* The error message is only supposed to describe how to fix the `arrow-body-style` problem; as usual, a fix could conflict with another rule.
* This is just a message for users and not an autofix, so we can assume users are willing to make some changes to the fix so that it fits within their code style.
* In a project that doesn't use the `undefined` identifier, I assume users would be familiar with the alternatives for using the `undefined` identifier, so they would be able to create an appropriate replacement for the expression.

Reply:
I like your alternative message suggestion. Ultimately this isn't a hill I want to die on, so just let me know what you decide and if you need me to review again. Thanks!

Reply:
Updated to change the message to:

> Unexpected block statement surrounding arrow body; put a value of `undefined` immediately after the `=>`.